### ISSUE TITLE
Slack integration double notes

### ISSUE THREAD

**Issue by sinteur** [RocketChat/Rocket.Chat#2439]
Using slack integration from gitlabs often repeats a message multiple times in the target channel

**Comment by sinteur** [comment]
Disabled the slack integration in gitlabs, messages kept coming in.

restarted gitlabs and chat, messages kept coming in

deleted the integration in rocket.chat, mesages finally stopped.

**Comment by rodrigok** [comment]
If a disabled integration in GitLab continues sending data to Rocket.Chat, seems that this problem is more related with GitLab then with Rocket.Chat.

**Comment by sinteur** [comment]
I'm sorry I wasn't as complete in the description above as I maybe should have been. This is what I did: stop gitlabs - message still coming in, so gitlabs can't be the source. restart gitlabs because this is a production environment. Stop and start rocket chat, still messages coming in. Delete integration in rocket chat, messages gone. Create a new integration, with resulting new token, messages still gone. Enter new integration token in gitlabs, enable it again, messages now coming in as single messages again.

**Comment by sinteur** [comment]
I may have a lead: it's possible that using the web integration token for a different channel is the cause

**Comment by engelgabriel** [comment]
Thanks for the extra info, but it looks like you have configured the same token in two gitlab different web gitlab hooks... please reopen this issue if the problem happens again.

I honestly can't understand how did you stop gitlabs and the message still coming in... what messages? Old messages being re posted?

**Comment by sinteur** [comment]
Yes, old messages being reposted. Note that we're effectively still talking about a but here because messages with the wrong token for a channel should simply be discarded - and I said "a lead" and not "a cause" because somewhere in the system these messages hang around without being marked delivered or thrown away until I actively delete the integration with the token that was used. The whole handling of the incoming slack messages need to be rethought.

**Comment by Cheaterman** [comment]
Same happens here, no idea why - worth noting we're using an outdated GitLab which for some reason doesn't provide the header defining which action was just taken, so I had to use request.content.object_kind in the switch.

FWIW, we send the notification to two different channels separated with a comma (published on channel #general,#otherchannel), but apparently sending to a single one doesn't help.

Maybe I should try entirely re-creating the hook? We already did that, no success.
